snapd-glib/snapd-json.c

```diff
@@ -165,14 +165,12 @@ snapd_json_get_date_time (JsonObject *object, const gchar *name)
 }
 
 static void
-parse_error_response (JsonObject *root, GError **error)
+parse_error_response (int status_code, JsonObject *root, GError **error)
 {
     const gchar *kind = NULL, *message = NULL;
-    gint64 status_code;
     JsonObject *result;
 
     result = snapd_json_get_object (root, "result");
-    status_code = snapd_json_get_int (root, "status-code", 0);
```

Comment:
Did you stop using the status code as provided in the JSON for a reason? I used this one over the HTTP one because I think the intention upstream is for the JSON responses to be potentially sent using other methods (e.g. websockets).